asmcomp/power/emit.mlp

```diff
@@ -429,36 +371,20 @@ module BR = Branch_relaxation.Make (struct
 
   let offset_pc_at_branch = 1
 
-  let size =
-    match abi with
-    | ELF32 -> (fun a _ _ -> a)
-    | ELF64v1 -> (fun _ b _ -> b)
-    | ELF64v2 -> (fun _ _ c -> c)
-
-  let profiling_prologue_size () =
-    match abi with
-    | ELF32 -> 5
-    | ELF64v1 | ELF64v2 -> 6
+  let profiling_prologue_size = 6
 
   let prologue_size f =
-    profiling_prologue_size ()
+    profiling_prologue_size
       + (if initial_stack_offset f > 0 then 1 else 0)
-      + (if f.fun_contains_calls then
-           2 +
-             match abi with
-             | ELF32 -> 0
-             | ELF64v1 | ELF64v2 -> 1
-         else 0)
+      + (if f.fun_contains_calls then 3 else 0)
 
   let tocload_size() =
     if !big_toc || !Clflags.for_package <> None then 2 else 1
```

Comment:
Is this also a good time to get rid of small TOC support and simplify code generation? Linux now defaults to `-mcmodel=medium`, large-toc has been the default in OCaml 4 for a while, and it seems unlikely anyone on 64-bit POWER would need TOC<64Kb.

I've pushed a diff that does this to https://github.com/avsm/ocaml/commit/25e7a1d9d8642b8041035783a29712cd37bf3be5 in case its helpful.